packages/api/endpoints/collections.js

```diff
@@ -84,32 +114,42 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
+  const {
+    collectionsModel = new models.Collection(),
+    dbClient = await db.connection.knex(),
+    logger = log,
+  } = req.testContext || {};
+
+  const collection = req.body || {};
+  const { name, version } = collection;
+
+  if (!name || !version) {
+    return res.boom.badRequest('Field name and/or version is missing');
+  }
+
+  if (await collectionsModel.exists(name, version)) {
+    return res.boom.conflict(`A record already exists for ${name} version: ${version}`);
+  }
+
   try {
-    const data = req.body;
-    const name = data.name;
-    const version = data.version;
+    const dynamoRecord = await collectionsModel.create(
+      omit(collection, 'dataType')
+    );
 
-    // make sure primary key is included
-    if (!name || !version) {
-      return res.boom.badRequest('Field name and/or version is missing');
-    }
-    const c = new models.Collection();
+    const dbRecord = dynamoRecordToDbRecord(dynamoRecord);
 
     try {
-      await c.get({ name, version });
-      return res.boom.conflict(`A record already exists for ${name} version: ${version}`);
+      await dbClient('collections').insert(dbRecord, 'cumulusId');
     } catch (error) {
-      if (error instanceof RecordDoesNotExist) {
-        await c.create(data);
+      await collectionsModel.delete({ name, version });
```

Comment:
I'm probably missing something obvious, but why are we deleting from Dynamo when we catch an error in a POST endpoint?

Reply:
This is handling the case where writing to Dynamo succeeded, but writing to RDS failed. In that case, we want to roll back the write to Dynamo and then throw the error.